c++/src/ColumnWriter.cc

```diff
@@ -139,10 +140,22 @@ namespace orc {
                          uint64_t offset,
                          uint64_t numValues,
                          const char* incomingMask) {
-    notNullEncoder->add(batch.notNull.data() + offset, numValues, incomingMask);
+    const char* notNull = batch.notNull.data() + offset;
+    notNullEncoder->add(notNull, numValues, incomingMask);
+    for (uint64_t i = 0; i < numValues; ++i) {
+      if (!notNull[i]) {
+        hasNullValue = true;
```

Comment:
I'd suggest checking hasNullValue in line 145 or 146 because hasNullValue may already be set by previous nulls.

Reply:
In addition, we should check batch.hasNulls as a shortcut.

Reply:
This logic has been changed to take the or operation with batch.hasNulls.